encodings/fastlanes/src/delta/compress.rs

```diff
@@ -99,13 +112,25 @@ where
 pub fn delta_decompress(array: DeltaArray) -> VortexResult<PrimitiveArray> {
     let bases = array.bases().into_primitive()?;
     let deltas = array.deltas().into_primitive()?;
-    let decoded = match_each_unsigned_integer_ptype!(deltas.ptype(), |$T| {
+
+    let unsigned_ptype = if bases.ptype().is_signed_int() {
+        bases.ptype().to_unsigned()
+    } else {
+        bases.ptype()
+    };
+
+    let decoded = match_each_unsigned_integer_ptype!(unsigned_ptype, |$T| {
         PrimitiveArray::from_vec(
-            decompress_primitive::<$T>(bases.maybe_null_slice(), deltas.maybe_null_slice()),
+            decompress_primitive::<$T>(
+                bases.reinterpret_cast(unsigned_ptype).maybe_null_slice(),
+                deltas.reinterpret_cast(unsigned_ptype).maybe_null_slice(),
+            ),
             array.validity()
         )
     });
-    Ok(decoded)
+    decoded
+        .slice(array.offset(), decoded.len() - array.trailing_garbage())?
```

Comment:
You can use the original array length here that will be only the valid values, i.e. offset + len